llvm/lib/Target/RISCV/RISCVVLOptimizer.cpp

```diff
@@ -257,17 +257,31 @@ static OperandInfo getOperandInfo(const MachineOperand &MO,
   // Vector Unit-Stride Instructions
   // Vector Strided Instructions
   /// Dest EEW encoded in the instruction and EMUL=(EEW/SEW)*LMUL
+  case RISCV::VLE8_V:
   case RISCV::VSE8_V:
+  case RISCV::VLM_V:
+  case RISCV::VSM_V:
```

Comment:
Ooops sorry, I somehow only saw the changes from the second commit when reviewing this. 

Is the EEW of VLM_V not always 1?

Reply:
updated

Reply:
Do we have any tests for vlm_v/vsm_v?

I think the vle/vlse/vlsxe parts are a bit more straightforward, we could split the masked loads and stores off to a separate PR

Reply:
I think vlm/vsm is pretty straight forward. I've added tests.